source/Engine.cpp

```diff
@@ -554,6 +556,16 @@ void Engine::Step(bool isActive)
 	wasActive = isActive;
 	Audio::Update(center);
 
+	if(GameData::IsLoaded())
+	{
+		Track::GameState state = Track::GameState::IDLE;
+		// If there are hostile ships in the system, set the state to COMBAT.
+		if(hadHostiles)
+			state = Track::GameState::COMBAT;
```

Comment:
I'm wondering, will this fade in and out or will it give boss music for a sparrow and then stop suddenly?

Reply:
Its transitions need work; this is a feature gap which should be fixed in a future enhancement

Reply:
fair, it probably isnt as simple as adding a time before it activates and then a cooldown before it can go off